hopp/simulation/technologies/hydrogen/h2_storage/lined_rock_cavern/lined_rock_cavern.py

```diff
@@ -0,0 +1,158 @@
+            self.installed_capex = self.lined_rock_cavern_storage_capex_per_kg * self.H2_storage_kg
+            cepci_overall = 1.29/1.30 # Convert from $2019 to $2018
+            self.installed_capex = cepci_overall * self.installed_capex
+            self.output_dict['lined_rock_cavern_storage_capex'] = self.installed_capex
+
+            outlet_pressure = 200 # Max outlet pressure of lined rock cavern in [1]
+            n_compressors = 2
+            storage_compressor = Compressor(outlet_pressure,self.system_flow_rate,n_compressors=n_compressors)
+            storage_compressor.compressor_power()
+            motor_rating, power = storage_compressor.compressor_system_power()
+            if motor_rating > 1600:
+                n_compressors += 1
+                storage_compressor = Compressor(outlet_pressure,self.system_flow_rate,n_compressors=n_compressors)
+                storage_compressor.compressor_power()
+                motor_rating, power = storage_compressor.compressor_system_power()
+            comp_capex,comp_OM = storage_compressor.compressor_costs()
+            cepci = 1.36/1.29 # convert from $2016 to $2018
+            self.comp_capex = comp_capex*cepci
+        elif self.model == 'hdsam':
+            raise NotImplementedError
+        return self.lined_rock_cavern_storage_capex_per_kg, self.installed_capex, self.comp_capex
+
+    def lined_rock_cavern_opex(self):
+        """
+        - Calculates the operation and maintenance costs excluding electricity costs for the lined rock cavern hydrogen storage
+        - Returns:
+            - `total_opex`: float - the OPEX (annual, fixed) in 2018 excluding electricity costs [USD/kg-yr]
+            - `output_dict` (dict):
+                - `'lined_rock_cavern_storage_opex'`: float - OPEX (annual, fixed) in 2018  [USD/yr]
+        """
+        # Operations and Maintenace costs [3]
+        # Labor 
+        # Base case is 1 operator, 24 hours a day, 7 days a week for a 100,000 kg/day average capacity facility.  Scaling factor of 0.25 is used for other sized facilities
+        annual_hours = 8760 * (self.system_flow_rate/100000)**0.25
+        self.overhead = 0.5 
+        labor = (annual_hours*self.labor_rate) * (1+self.overhead) # Burdened labor cost
+        insurance = self.insurance * self.installed_capex
+        property_taxes = self.property_taxes * self.installed_capex
+        licensing_permits = self.licensing_permits * self.installed_capex
+        comp_op_maint = self.comp_om * self.comp_capex 
+        facility_op_maint = self.facility_om * (self.installed_capex - self.comp_capex)
+
+        # O&M excludes electricity requirements
+        total_om = labor+insurance+licensing_permits+property_taxes+comp_op_maint+facility_op_maint
+        self.output_dict['lined_rock_cavern_storage_opex'] = total_om
+        return total_om
+
+if __name__ == '__main__':
```

Comment:
Unless this is intended to run as a one-off script, we probably don't need this block